.travis.yml

```diff
@@ -19,15 +19,46 @@ env:
         # Make sure beaver is in the PATH
         - PATH="$(git config -f .gitmodules submodule.beaver.path)/bin:$PATH"
     matrix:
-        - DMD=1.081.0 DIST=xenial F=production AFTER_SCRIPT=1
+        - DMD=1.081.0 DIST=xenial F=production
         - DMD=1.081.0 DIST=xenial F=devel
         - DMD=2.070.2.s12 DIST=xenial F=production
         - DMD=2.070.2.s12 DIST=xenial F=devel
         - DMD=2.071.2.s1 DIST=xenial F=production
         - DMD=2.071.2.s1 DIST=xenial F=devel
 
-install: beaver dlang install
+install: |
+    zcat "build/last/tmp/docker.img.gz" |
+        /usr/bin/time -f 'Load took %es\n' docker load -q || true
+    beaver dlang install
 
-script: beaver dlang make
+script: /usr/bin/time -f 'Build took %es\n' beaver dlang make
 
-after_success: ci/travis-after-success.sh
+after_success: |
+    # Write the image cache
+    docker images --all --format '{{.Repository}}:{{.Tag}} {{.ID}}' |
+        sed 's|<none>:<none>||g' |
+        /usr/bin/time -f 'Save took %es\n' xargs docker save |
+        gzip --fast > "build/last/tmp/docker.img.gz"
+
+jobs:
+    include:
+        - stage: Coverage
```

Comment:
What will happen if you also define `stage: Test` and move env matrix there? Will it work as I would hope or fail miserably? :)

Reply:
I think it should work, but then we'll have to redefine the `install:` step for all the stages, which is just annoying. Everything you don't put in a stage goes to the default `Test` stage anyway.